apps/condo/domains/acquiring/schema/Payment.js

```diff
@@ -22,22 +42,37 @@ const Payment = new GQLListSchema('Payment', {
         sender: SENDER_FIELD,
 
         amount: {
-            ...MONEY_AMOUNT_FIELD,
+            ...POSITIVE_MONEY_AMOUNT_FIELD,
             schemaDoc: 'Amount of money from MultiPayment.amountWithOutExplicitFee to pay for billing receipt',
             isRequired: true,
         },
 
         explicitFee: {
-            ...MONEY_AMOUNT_FIELD,
+            ...NON_NEGATIVE_MONEY_FIELD,
             schemaDoc: 'Amount of money which payer pays on top of initial "amount"',
             isRequired: false,
         },
 
         implicitFee: {
-            ...MONEY_AMOUNT_FIELD,
+            ...NON_NEGATIVE_MONEY_FIELD,
             schemaDoc: 'Amount of money which recipient pays from initial amount for transaction',
             isRequired: false,
             access: { read: access.canReadPaymentsSensitiveData },
+            hooks: {
+                validateInput: ({ resolvedData, addFieldValidationError, fieldPath, listKey, operation, existingItem }) => {
```

Comment:
you should understand, that you overwrite the `NON_NEGATIVE_MONEY_FIELD` validation hook. Is it ok?

Reply:
yes, since I expand it with custom logic

Reply:
replaced with SignedDecimal + custom validation